packages/ts/generator-typescript-core/src/Generator.ts

```diff
@@ -13,11 +13,13 @@ export default class Generator {
   readonly #logger: LoggerFactory;
   readonly #manager: PluginManager;
   readonly #parser: SwaggerParser;
+  readonly #outputDir: string | undefined;
 
-  public constructor(plugins: readonly PluginConstructor[], logger: LoggerFactory) {
+  public constructor(plugins: readonly PluginConstructor[], context: { logger: LoggerFactory; outputDir?: string }) {
```

Comment:
It's better to have named types instead of object literals. Since it is a Generator context, let's delcare it just above the `Generator` class:
```ts
export type GeneratorContext = Readonly<{
  logger: LoggerFactory;
  outputDir?: string;
}>;
```